src/script/descriptor.cpp

```diff
@@ -212,6 +215,18 @@ class OriginPubkeyProvider final : public PubkeyProvider
         ret = "[" + OriginString() + "]" + std::move(sub);
         return true;
     }
+    bool ToNormalizedString(const SigningProvider& arg, std::string& ret, bool priv) const override
+    {
+        std::string sub;
+        if (!m_provider->ToNormalizedString(arg, sub, priv)) return false;
+        if (sub[0] == '[') {
+            sub = sub.substr(10);
```

Comment:
I'm not sure I fully understand this part. From what I figured here you're handling a case when we wrap another `OriginPubkeyProvider`. If so, I have a few questions:

1) What do we even expect to happen in such case? I can't think of realistic scenario
2) Why handle it here if we don't handle it in other methods, like `ToPrivateString` or `ToString()`
3) Why do you think the origin from wrapped provider is exactly `10` chars long? Even if so, than there is a closing bracket missing

Hope my questions make at least some sense

Reply:
The goal of `ToNormalizedString` is to create a descriptor that can be imported into another wallet. A descriptor might not be able to be imported because hardened derivation is in use. So for `BIP32PubkeyProvider::ToNormalizedString`, we are deriving the key at the last hardened step and returning that key. As we did additional derivation not present in the original descriptor, we are generating new origin info that needs to be combined with any existing origin info.

1. We don't actually wrap another `OriginPubkeyProvider`. Rather we might be wrapping a `BIP32PubkeyProvider` where it's `ToNormalizedString` might return us a string containing origin info as it did additional derivation.
2. `ToPrivateString` and `ToString` return the descriptor as-is, no additional derivation is done. Thus there is no need to handle this "origin within the origin provider."
3. This is to cut off the first 10 characters because we don't need them. Specifically, that's the `[` followed by 8 characters for the fingerprint, followed by the first `/`. Actually, there might be an off-by-one here as I think we need that `/`.

Reply:
I created and verified a test case to cover this part of code and which would have exposed the bug as well.
Feel free to add it.

```cpp
Check(
"pkh([01234567/10/20]xprv9s21ZrQH143K31xYSDQpPDxsXRTUcvj2iNHm5NUtrGiGG5e2DtALGdso3pGz6ssrdK4PFmM8NSpSBHNqPqm55Qn3LqFtT2emdEXVYsCzC2U/2147483647'/0)",
"pkh([01234567/10/20]xpub661MyMwAqRbcFW31YEwpkMuc5THy2PSt5bDMsktWQcFF8syAmRUapSCGu8ED9W6oDMSgv6Zz8idoc4a6mr8BDzTJY47LJhkJ8UB7WEGuduB/2147483647'/0)",
"pkh([01234567/10/20/2147483647']xprv9vHkqa6XAPwKqSKSEJMcAB3yoCZhaSVsGZbSkFY5L3Lfjjk8sjZucbsbvEw5o3QrSA69nPfZDCgFnNnLhQ2ohpZuwummndnPasDw2Qr6dC2/0)",
"pkh([01234567/10/20/2147483647']xpub69H7F5dQzmVd3vPuLKtcXJziMEQByuDidnX3YdwgtNsecY5HRGtAAQC5mXTt4dsv9RzyjgDjAQs9VGVV6ydYCHnprc9vvaA5YtqWyL6hyds/0)",
HARDENED,
{{"76a914ebdc90806a9c4356c1c88e42216611e1cb4c1c1788ac"}},
OutputType::LEGACY,
{{10,20,0xFFFFFFFFUL,0}}
);
```

Reply:
Added this test.